virtio-devices/src/mem.rs

```diff
@@ -724,54 +724,59 @@ impl MemEpollHandler {
 }
 
 impl EpollHelperHandler for MemEpollHandler {
-    fn handle_event(&mut self, _helper: &mut EpollHelper, event: &epoll::Event) -> bool {
+    fn handle_event(
+        &mut self,
+        _helper: &mut EpollHelper,
+        event: &epoll::Event,
+    ) -> result::Result<(), EpollHelperError> {
         let ev_type = event.data as u16;
         match ev_type {
             RESIZE_EVENT => {
-                if let Err(e) = self.resize.evt.read() {
-                    error!("Failed to get resize event: {:?}", e);
-                    return true;
-                } else {
-                    let size = self.resize.size();
-                    let mut config = self.config.lock().unwrap();
-                    let mut signal_error = false;
-                    let mut r = config.resize(size);
-                    r = match r {
-                        Err(e) => Err(e),
-                        _ => match self.signal(VirtioInterruptType::Config) {
-                            Err(e) => {
-                                signal_error = true;
-                                Err(Error::ResizeTriggerFail(e))
-                            }
-                            _ => Ok(()),
-                        },
-                    };
-                    if let Err(e) = self.resize.send(r) {
```

Comment:
Same here, can you do the conversion first and then in a separate commit do the improvement.